src/Regridder.jl

```diff
@@ -633,5 +633,52 @@ function cgll2latlonz(field; DIR = "cgll2latlonz_dir", nlat = 360, nlon = 720, c
     return new_data, coords
 end
 
+"""
+    truncate_dataset(date0, timeStart, timeEnd, datafile)
+
+Truncates given data set, and constructs a new dataset containing only the dates that are used in the simulation
+"""
+function truncate_dataset(date0, time_start, time_end, datafile, datapath)
+    date_start = date0 + Dates.Second(time_start)
+    date_end = date0 + Dates.Second(time_start + time_end)
+
+    ds = NCDataset(datafile, "r")
+    dates = ds["time"][:]
+
+    if date_start < dates[1]        # if the simulation start date is before our first date in the dataset, we leave the beginning of the truncated dataset to be the first date available
+        start_id = 1
+    elseif date_start > last(dates)     # if the simulation start date is after the last date in the dataset, we start the truncated dataset at its last possible date
+        start_id = length(dates)
+    else                                # if the simulation start date falls within the range of the dataset, we find the closest date to the start date and truncate there 
+        (~, start_id) = findmin(x -> abs(x - date_start), dates)
+        if dates[start_id] > date_start     # if the closest date is after the start date, we add one more date before it to the dataset 
+            start_id = start_id - 1
+        end
+    end
+
+    if date_end < dates[1]              # if the simulation end date is before our first date in the dataset, we truncate the end of the dataset to be the first date
+        end_id = 1
+    elseif date_end > last(dates)       # if the simulation end date is after the last date in the dataset, we leave the end of the dataset as is 
+        end_id = length(dates)
+    else                                # if the simulation end date falls within the range of the dataset, we find the closest date to the end date and truncate there 
+        (~, end_id) = findmin(x -> abs(x - date_end), dates)
+        if dates[end_id] < date_end         # if the closest date is before the end date, we add one more date after it to the dataset 
+            end_id = end_id + 1
+        end
+    end
+
+    file_name = string("truncated_data_", string(date_start), string(date_end), ".nc")
+    datafile_truncated = joinpath(datapath, file_name)
+    truncated_ds = NCDataset(datafile_truncated, "c")
```

Comment:
This line is producing errors in buildkite: `ERROR: LoadError: NetCDF error: Permission denied (NetCDF error code: 13)`. From searching the error code online, it seems like this means either the file exists but we don't have permission to write to it, or it doesn't exist but we don't have permission to create new files in that directory. It would be good to figure out which of these two cases we're in. It sounds like you were able to run this locally, so maybe we don't have write permission for the folder on the cluster where we try to create the file in buildkite runs.

Reply:
Would this overwrite all of the files to the same path, since they would all be in REGRID_DIR.truncated_data....nc?

Reply:
you should be able to set up a `run_name` specific path. The point was to write this somewhere outside the `.depot` directory.